mkosi

```diff
@@ -2023,6 +2023,31 @@ def install_arch(args: CommandLineArguments, workspace: str, do_run_build_script
     os.makedirs(os.path.join(root, "var/lib/pacman"), 0o755, exist_ok=True)
     os.makedirs(os.path.join(root, "etc/pacman.d/gnupg"), 0o755, exist_ok=True)
 
+    # Permissions on these directories are all 0o777 because of `mount --bind`
+    # limitations but pacman expects them to be 0o755 so we fix them before
+    # calling pacstrap (except /var/tmp which is 0o1777).
+    fix_permissions_dirs = [
+        ["boot", 0o755],
+        ["etc", 0o755],
+        ["etc/pacman.d", 0o755],
+        ["var", 0o755],
+        ["var/lib", 0o755],
+        ["var/cache", 0o755],
+        ["var/cache/pacman", 0o755],
+        ["var/tmp", 0o1777],
+    ]
+
+    for dir, permissions in fix_permissions_dirs:
+        path = os.path.join(root, dir)
+        if os.path.exists(path):
+            os.chmod(path, permissions)
+
+    # Disable mkinitcpio hook because we manually install the kernel image to
+    # /boot using kernel-install in install_boot_loader_arch.
+    hooks = os.path.join(root, "etc/pacman.d/hooks")
+    os.makedirs(hooks, 0o755, exist_ok=True)
+    os.symlink("/dev/null", os.path.join(hooks, "90-mkinitcpio-install.hook"))
```

Comment:
don't you want to undo this when you are done? Otherwise the image produced is not so useful as this hook will remain disabled (see similar issue solved for fedora

Reply:
Good point, I'll make a new PR that removes this after pacman has finished.